databricks/koalas/frame.py

```diff
@@ -1963,6 +1963,68 @@ def sort_values(self, by: Union[str, List[str]], ascending: Union[bool, List[boo
         else:
             return kdf
 
+    def sort_index(self, axis: int = 0, level: int = None, ascending: bool = True,
+                   inplace: bool = False, kind: str = None, na_position: str = 'last') \
+            -> Optional['DataFrame']:
+        """
+        Sort object by labels (along an axis)
+
+        Parameters
+        ----------
+        axis : index, columns to direct sorting
+        level : int or level name or list of ints or list of level names
+            if not None, sort on values in specified index level(s)
+        ascending : boolean, default True
+            Sort ascending vs. descending
+        inplace : bool, default False
+            if True, perform operation in-place
+        kind : Koalas does not allow specifying the sorting algorithm at the moment, default None
+        na_position : {‘first’, ‘last’}, default ‘last’
+            first puts NaNs at the beginning, last puts NaNs at the end. Not implemented for
+            MultiIndex.
+
+        Returns
+        -------
+        sorted_obj : DataFrame
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': [2, 1, np.nan]}, index=['b', 'a', np.nan])
+
+        >>> df.sort_index()
+               A
+        a    1.0
+        b    2.0
+        NaN  NaN
+
+        >>> df.sort_index(ascending=False)
+               A
+        b    2.0
+        a    1.0
+        NaN  NaN
+
+        >>> df.sort_index(na_position='first')
+               A
+        NaN  NaN
+        a    1.0
+        b    2.0
+
+        >>> df.sort_index(inplace=True)
+        >>> df
+               A
+        a    1.0
+        b    2.0
+        NaN  NaN
+        """
+        if axis != 0:
+            raise ValueError("No other axes than 0 are supported at the moment")
+        if level is not None:
+            raise ValueError("The 'axis' argument is not supported at the moment")
+        if kind is not None:
+            raise ValueError("Specifying the sorting algorithm is supported at the moment.")
+        return self.sort_values(by='__index_level_0__', ascending=ascending, inplace=inplace,
```

Comment:
The actual index column names are in the metadata. We should use it: `by=self._metadata.index_columns`.

Reply:
I didn't know that. Thanks a lot for pointing it out! I'm sure I've also used `'__index_level_0__'` in some other PRs so I'll make sure to fix it there as well but for now, 46fbd2e fixes it for this PR.

Reply:
If the PR in your mind is the `merge` one, I'm working on it with some improvements, otherwise please go ahead. Thanks!

Reply:
Thanks for letting me know. I've just done a text search across the project and `merge` seems to be the only other place where '__index_level_0__' appears right now, except for the `meta` module.